app/src/main/java/com/nervousfish/nervousfish/modules/pairing/AndroidBluetoothConnectThread.java

```diff
@@ -44,54 +42,71 @@
         } catch (final IOException e) {
             LOGGER.error("Connection failed", e);
         }
-        socket = tmp;
+        this.socket = tmp;
         serviceLocator.postOnEventBus(new BluetoothConnectingEvent());
+        this.thread = new Thread(new AndroidBluetoothConnectThread.ConnectThread(serviceLocator, this.socket));
     }
 
     /**
-     * {@inheritDoc}
+     * Starts the thread.
      */
-    @Override
-    public void run() {
-        LOGGER.info("Connect Bluetooth thread started");
-        setName("AndroidBluetoothConnectThread thread");
-
-        // Always cancel discovery because it will slow down a connection
-        BluetoothAdapter.getDefaultAdapter().cancelDiscovery();
-
-        // Make a connection to the BluetoothSocket
-        try {
-            // This is a blocking call and will only return on a
-            // successful connection or an exception
-            socket.connect();
-        } catch (final IOException e) {
-            LOGGER.warn("Exception while connecting over Bluetooth", e);
-            try {
-                socket.close();
-            } catch (final IOException esocketCloseException) {
-                LOGGER.error("Connection failed/couldn't close the socket", esocketCloseException);
-            }
-            serviceLocator.postOnEventBus(new BluetoothConnectionFailedEvent());
-            return;
-        }
-
-        serviceLocator.postOnEventBus(new BluetoothAlmostConnectedEvent(socket));
+    void start() {
+        this.thread.start();
     }
 
     /**
-     * Cancels the connect thread and closes the socket if closeSocket is set to true
+     * Cancels the connect thread and optionally closes the socket
      *
-     * @param closeSocket True if its socket should be closed.
+     * @param closeSocket True if the socket should be closed
      */
     void cancel(final boolean closeSocket) {
         LOGGER.warn("Cancelled!");
         if (closeSocket) {
             try {
-                socket.close();
+                this.socket.close();
                 LOGGER.warn("Socket closed!");
             } catch (final IOException e) {
                 LOGGER.error("Closing socket", e);
             }
         }
     }
+
+    private static final class ConnectThread implements Runnable {
```

Comment:
Yeah, because StackOverflow and Intellij suggest that it's better to implement Runnable rather than extend Thread, because you can only extend a single class but you can implement multiple other classes